src/rpc/dial.rs

```diff
@@ -792,16 +790,21 @@ async fn maybe_connect_via_webrtc(
     let sent_done_or_error = Arc::new(AtomicBool::new(false));
     let uuid_lock = Arc::new(RwLock::new("".to_string()));
     let uuid_for_ice_gathering_thread = uuid_lock.clone();
-    let is_open = Arc::new(AtomicBool::new(false));
-    let is_open_read = is_open.clone();
+
+    // Using an mpsc channel to report unrecoverable errors during Signaling, so we
+    // don't have to wait until the timeout expires before giving up on this attempt.
+    // The size of the channel is set to 1 since any error (or success) should terminate the function
+    let (is_open_s, mut is_open_r) = mpsc::channel(1);
```

Comment:
If it's an `mpsc` channel with buffer size 1, why not use an [`oneshot` channel](https://docs.rs/tokio/latest/tokio/sync/index.html#oneshot-channel)?

Reply:
oneshot channels cannot be cloned or reused multiple time (send() takes ownership) We could work around it with Arc and interior mutability but it is a bit less elegant IMO